packages/react-client/src/DeviceManager.ts

```diff
@@ -130,22 +130,24 @@ export class DeviceManager
     this.storageConfig?.saveLastDevice(info);
   }
 
-  // todo `audioDeviceId / videoDeviceId === true` means use last device
-  public async start(deviceId?: string | boolean) {
-    const shouldRestart = !!deviceId && deviceId !== this.deviceState.media?.deviceInfo?.deviceId;
+  public async start(deviceId?: string) {
+    const newDeviceId = deviceId ?? this.getLastDevice()?.deviceId ?? true; // true means start any available device
+    const currentDeviceId = this.deviceState.media?.deviceInfo?.deviceId;
 
-    const newDevice = deviceId === true ? this.getLastDevice()?.deviceId || true : deviceId;
+    const shouldReplaceDevice = Boolean(currentDeviceId && currentDeviceId !== newDeviceId);
+    const isDeviceStopped = currentDeviceId === undefined;
+    const shouldProceed = isDeviceStopped || shouldReplaceDevice;
 
     const trackConstraints = this.constraints;
 
-    const exactConstraints = shouldRestart && prepareMediaTrackConstraints(newDevice, trackConstraints);
+    const exactConstraints = shouldProceed && prepareMediaTrackConstraints(newDeviceId, trackConstraints);
     if (!exactConstraints) return;
```

Comment:
I would consider refactoring it to separate function. `deviceId` being `true` for some random reason is... this just should not be written like that.

Reply:
You're right, fixed!